src/encoded/static/libs/jsonld.js

```diff
@@ -0,0 +1,146 @@
+            name: 'Stanford Univeristy',
+        },
+    };
+};
+
+const _mapContextToDistribution = (context, measurementTechnique) => {
+    if (!context) {
+        return {};
+    }
+
+    const distribution = {
+        name: context.accession,
+        contentUrl: `${baseUrl}${context['@id']}`,
+        sameAs: `${baseUrl}${context['@id']}`,
+        url: `${baseUrl}${context['@id']}`,
+        identifier: context.uuid,
+        encodingFormat: 'text',
+    };
+
+    if (measurementTechnique) {
+        distribution.measurementTechnique = measurementTechnique;
+    }
+
+    return distribution;
+};
+
+const _mapSourceToSourceOrganization = (source) => {
+    if (!source) {
+        return {};
+    }
+
+    return {
+        name: source.title,
+        sameAs: source.url,
+        description: source.description,
+        url: `${baseUrl}${source['@id']}`,
+        identifier: source.uuid,
+    };
+};
+
+const _mapSubmitterToCreator = (submittedBy) => {
+    if (!submittedBy) {
+        return {};
+    }
+
+    return {
+        '@type': 'Person',
+        identifier: submittedBy.uuid,
+        name: submittedBy.title,
+        worksFor: {
+            '@type': 'EducationalOrganization',
+            sameAs: `${baseUrl}${submittedBy.lab}`,
+            url: `${baseUrl}${submittedBy.lab}`,
+        },
+        memberOf: !submittedBy.submits_for ? [] : submittedBy.submits_for.map(submitsFor => ({
+            '@type': 'EducationalOrganization',
+            url: `${baseUrl}${submitsFor}`,
+        })),
+    };
+};
+
+const jsonldFormatter = (context, url) => {
+    if (!context) {
+        return {};
+    }
+
+    baseUrl = url || baseUrl;
+    const isAnnotationDataSet = context['@type'].some(type => type.toLowerCase() === 'annotation');
+    const measurementTechnique = !isAnnotationDataSet ? `${context.assay_title} (${context.assay_term_id})` : '';
+
+    const mappedData = {
+        '@context': 'http://schema.org/',
+        '@type': 'Dataset',
+        url: `${baseUrl}${context['@id']}`,
+        dateCreated: context.date_created,
+        license: `${baseUrl}/help/citing-encode/`,
+        name: context.accession, // required by Google data set
+        isAccessibleForFree: true,
+        isFamilyFriendly: true,
+        identifier: context.uuid,
+        version: context.schema_version,
+        publisher: _addPublisher(context),
+        distribution: _mapContextToDistribution(context, measurementTechnique),
+        keywords: [...context.internal_tags, ...[context.accession, context.uuid, 'ENCODE', 'ENCODED', 'DCC', 'Encyclopedia of DNA Elements']],
+        measurementTechnique,
+        includedInDataCatalog: {
+            '@type': 'DataCatalog',
+            name: 'ENCODE: Encyclopedia of DNA Elements',
+            url: `${baseUrl}`,
+        },
+    };
+
+    const partialDescription = [context.award.project, context.award.name, context.lab.title].filter(d => d !== '').join(' - ');
```

Comment:
I think technically you don’t need the filtering step, as award and lab are required in all three dataset types, and project and name are required in award, and title is required in lab.